Fails to build with Clang + -Werror

Applying https://github.com/phkaeser/libbase/commit/349d02e9cabb to wlmaker produces:
```c
$ export CC=clang CXX=clang++
$ cmake -B /tmp/wlmaker_build
$ cmake --build /tmp/wlmaker_build
[...]
src/clip.c:330:14: error: explicitly assigning value of variable of type 'wlmaker_clip_t *' (aka 'struct _wlmaker_clip_t *') to itself [-Werror,-Wself-assign]
  330 |     clip_ptr = clip_ptr;  // unused.
      |     ~~~~~~~~ ^ ~~~~~~~~
src/clip.c:377:21: error: explicitly assigning value of variable of type 'wlmaker_interactive_t *' (aka 'struct _wlmaker_interactive_t *') to itself [-Werror,-Wself-assign]
  377 |     interactive_ptr = interactive_ptr;  // unused.
      |     ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
src/clip.c:393:21: error: explicitly assigning value of variable of type 'wlmaker_interactive_t *' (aka 'struct _wlmaker_interactive_t *') to itself [-Werror,-Wself-assign]
  393 |     interactive_ptr = interactive_ptr;  // unused.
      |     ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
src/dock_app.c:312:27: error: explicitly assigning value of variable of type 'wlmaker_subprocess_handle_t *' (aka 'struct _wlmaker_subprocess_handle_t *') to itself [-Werror,-Wself-assign]
  312 |     subprocess_handle_ptr = subprocess_handle_ptr;
      |     ~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
src/iconified.c:204:27: error: explicitly assigning value of variable of type 'struct wlr_scene_surface *' to itself [-Werror,-Wself-assign]
  204 |     wlr_scene_surface_ptr = wlr_scene_surface_ptr;
      |     ~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
src/menu_item.c:95:19: error: explicitly assigning value of variable of type 'const wlmaker_menu_item_t *' (aka 'const struct _wlmaker_menu_item_t *') to itself [-Werror,-Wself-assign]
   95 |     menu_item_ptr = menu_item_ptr;  // currently unused.
      |     ~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~
src/layer_shell.c:135:23: error: explicitly assigning value of variable of type 'wlmaker_layer_surface_t *' (aka 'struct _wlmaker_layer_surface_t *') to itself [-Werror,-Wself-assign]
  135 |     layer_surface_ptr = layer_surface_ptr;
      |     ~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~
src/menu.c:238:14: error: explicitly assigning value of variable of type 'wlmaker_menu_t *' to itself [-Werror,-Wself-assign]
  238 |     menu_ptr = menu_ptr;
      |     ~~~~~~~~ ^ ~~~~~~~~
src/menu.c:259:14: error: explicitly assigning value of variable of type 'wlmaker_menu_t *' to itself [-Werror,-Wself-assign]
  259 |     menu_ptr = menu_ptr;
      |     ~~~~~~~~ ^ ~~~~~~~~
src/menu.c:260:7: error: explicitly assigning value of variable of type 'double' to itself [-Werror,-Wself-assign]
  260 |     x = x; y = y;
      |     ~ ^ ~
src/menu.c:260:14: error: explicitly assigning value of variable of type 'double' to itself [-Werror,-Wself-assign]
  260 |     x = x; y = y;
      |            ~ ^ ~
src/clip.c:330:14: error: explicitly assigning value of variable of type 'wlmaker_clip_t *' (aka 'struct _wlmaker_clip_t *') to itself [-Werror,-Wself-assign]
  330 |     clip_ptr = clip_ptr;  // unused.
      |     ~~~~~~~~ ^ ~~~~~~~~
src/clip.c:377:21: error: explicitly assigning value of variable of type 'wlmaker_interactive_t *' (aka 'struct _wlmaker_interactive_t *') to itself [-Werror,-Wself-assign]
  377 |     interactive_ptr = interactive_ptr;  // unused.
      |     ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
src/clip.c:393:21: error: explicitly assigning value of variable of type 'wlmaker_interactive_t *' (aka 'struct _wlmaker_interactive_t *') to itself [-Werror,-Wself-assign]
  393 |     interactive_ptr = interactive_ptr;  // unused.
      |     ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
src/dock_app.c:312:27: error: explicitly assigning value of variable of type 'wlmaker_subprocess_handle_t *' (aka 'struct _wlmaker_subprocess_handle_t *') to itself [-Werror,-Wself-assign]
  312 |     subprocess_handle_ptr = subprocess_handle_ptr;
      |     ~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
src/iconified.c:204:27: error: explicitly assigning value of variable of type 'struct wlr_scene_surface *' to itself [-Werror,-Wself-assign]
  204 |     wlr_scene_surface_ptr = wlr_scene_surface_ptr;
      |     ~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
src/menu_item.c:95:19: error: explicitly assigning value of variable of type 'const wlmaker_menu_item_t *' (aka 'const struct _wlmaker_menu_item_t *') to itself [-Werror,-Wself-assign]
   95 |     menu_item_ptr = menu_item_ptr;  // currently unused.
      |     ~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~
src/layer_shell.c:135:23: error: explicitly assigning value of variable of type 'wlmaker_layer_surface_t *' (aka 'struct _wlmaker_layer_surface_t *') to itself [-Werror,-Wself-assign]
  135 |     layer_surface_ptr = layer_surface_ptr;
      |     ~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~
src/menu.c:238:14: error: explicitly assigning value of variable of type 'wlmaker_menu_t *' to itself [-Werror,-Wself-assign]
  238 |     menu_ptr = menu_ptr;
      |     ~~~~~~~~ ^ ~~~~~~~~
src/menu.c:259:14: error: explicitly assigning value of variable of type 'wlmaker_menu_t *' to itself [-Werror,-Wself-assign]
  259 |     menu_ptr = menu_ptr;
      |     ~~~~~~~~ ^ ~~~~~~~~
src/menu.c:260:7: error: explicitly assigning value of variable of type 'double' to itself [-Werror,-Wself-assign]
  260 |     x = x; y = y;
      |     ~ ^ ~
src/menu.c:260:14: error: explicitly assigning value of variable of type 'double' to itself [-Werror,-Wself-assign]
  260 |     x = x; y = y;
      |            ~ ^ ~
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,8 +49,8 @@ OPTION(config_DEBUG "Include debugging information" ON)
 OPTION(config_OPTIM "Optimizations" OFF)
 OPTION(config_DOXYGEN_CRITICAL "Whether to fail on doxygen warnings" OFF)
 
-# Toplevel compile options, for GCC.
-IF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+# Toplevel compile options, for GCC and clang.
+IF(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
   ADD_COMPILE_OPTIONS(-Wall -Wextra -Werror)
 
   IF(config_DEBUG)
@@ -67,7 +67,7 @@ IF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
   # full path. This option resets it to a path relative to project source.
   ADD_COMPILE_OPTIONS(-fmacro-prefix-map=${PROJECT_SOURCE_DIR}=.)
 
-ENDIF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+ENDIF(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
 SET(CMAKE_C_STANDARD 11)
 
 LIST(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")

--- a/src/clip.c
+++ b/src/clip.c
@@ -322,12 +322,10 @@ wlmaker_clip_t *clip_from_view(wlmaker_view_t *view_ptr)
  * @param width_ptr
  * @param height_ptr
  */
-void clip_get_size(wlmaker_view_t *view_ptr,
+void clip_get_size(__UNUSED__ wlmaker_view_t *view_ptr,
                    uint32_t *width_ptr,
                    uint32_t *height_ptr)
 {
-    wlmaker_clip_t *clip_ptr = clip_from_view(view_ptr);
-    clip_ptr = clip_ptr;  // unused.
     if (NULL != width_ptr) *width_ptr = 64;
     if (NULL != height_ptr) *height_ptr = 64;
 }
@@ -368,13 +366,12 @@ void draw_workspace(cairo_t *cairo_ptr, int num, const char *name_ptr)
  * @param interactive_ptr
  * @param data_ptr            points to the `wlmaker_clip_t`.
  */
-void callback_prev(wlmaker_interactive_t *interactive_ptr,
+void callback_prev(__UNUSED__ wlmaker_interactive_t *interactive_ptr,
                    void *data_ptr)
 {
     wlmaker_clip_t *clip_ptr = (wlmaker_clip_t*)data_ptr;
 
     wlmaker_server_switch_to_previous_workspace(clip_ptr->server_ptr);
-    interactive_ptr = interactive_ptr;  // unused.
 }
 
 /* ------------------------------------------------------------------------- */
@@ -384,13 +381,12 @@ void callback_prev(wlmaker_interactive_t *interactive_ptr,
  * @param interactive_ptr
  * @param data_ptr            points to the `wlmaker_clip_t`.
  */
-void callback_next(wlmaker_interactive_t *interactive_ptr,
+void callback_next(__UNUSED__ wlmaker_interactive_t *interactive_ptr,
                    void *data_ptr)
 {
     wlmaker_clip_t *clip_ptr = (wlmaker_clip_t*)data_ptr;
 
     wlmaker_server_switch_to_next_workspace(clip_ptr->server_ptr);
-    interactive_ptr = interactive_ptr;  // unused.
 }
 
 /* ------------------------------------------------------------------------- */

--- a/src/dock_app.c
+++ b/src/dock_app.c
@@ -294,7 +294,7 @@ void tile_callback(
         return;
     }
 
-    wlmaker_subprocess_handle_t *subprocess_handle_ptr;
+    __UNUSED__ wlmaker_subprocess_handle_t *subprocess_handle_ptr;
     subprocess_handle_ptr = wlmaker_subprocess_monitor_entrust(
         dock_app_ptr->view_ptr->server_ptr->monitor_ptr,
         subprocess_ptr,
@@ -309,7 +309,6 @@ void tile_callback(
     // error status and permitting to kill the subprocess.
     // Note: There may be more than 1 subprocess for the launcher (possibly
     // depending on configuration.
-    subprocess_handle_ptr = subprocess_handle_ptr;
 }
 
 /* ------------------------------------------------------------------------- */

--- a/src/iconified.c
+++ b/src/iconified.c
@@ -197,11 +197,10 @@ void wlmaker_dockapp_iconified_attach(
     wlmaker_dockapp_iconified_t *dai_ptr,
     struct wlr_surface *wlr_surface_ptr)
 {
-    struct wlr_scene_surface *wlr_scene_surface_ptr = wlr_scene_surface_create(
-        dai_ptr->wlr_scene_tree_ptr,
-        wlr_surface_ptr);
-
-    wlr_scene_surface_ptr = wlr_scene_surface_ptr;
+    __UNUSED__ struct wlr_scene_surface *wlr_scene_surface_ptr =
+        wlr_scene_surface_create(
+            dai_ptr->wlr_scene_tree_ptr,
+            wlr_surface_ptr);
 }
 
 /* ------------------------------------------------------------------------- */

--- a/src/layer_shell.c
+++ b/src/layer_shell.c
@@ -128,11 +128,10 @@ void handle_new_surface(
             layer_shell_ptr->server_ptr);
     }
 
-    wlmaker_layer_surface_t *layer_surface_ptr = wlmaker_layer_surface_create(
-        wlr_layer_surface_v1_ptr,
-        layer_shell_ptr->server_ptr);
-    // not used further.
-    layer_surface_ptr = layer_surface_ptr;
+    __UNUSED__ wlmaker_layer_surface_t *layer_surface_ptr =
+        wlmaker_layer_surface_create(
+            wlr_layer_surface_v1_ptr,
+            layer_shell_ptr->server_ptr);
 }
 
 /* == End of layer_shell.c ================================================= */

--- a/src/menu.c
+++ b/src/menu.c
@@ -231,11 +231,8 @@ void _menu_motion(
  *
  * @param interactive_ptr
  */
-void _menu_focus(wlmaker_interactive_t *interactive_ptr)
+void _menu_focus(    __UNUSED__ wlmaker_interactive_t *interactive_ptr)
 {
-    wlmaker_menu_t *menu_ptr = menu_from_interactive(interactive_ptr);
-
-    menu_ptr = menu_ptr;
 }
 
 /* ------------------------------------------------------------------------- */
@@ -251,13 +248,11 @@ void _menu_focus(wlmaker_interactive_t *interactive_ptr)
  */
 void _menu_button(
     wlmaker_interactive_t *interactive_ptr,
-    double x, double y,
+    __UNUSED__ double x,
+    __UNUSED__ double y,
     struct wlr_pointer_button_event *wlr_pointer_button_event_ptr)
 {
     wlmaker_menu_t *menu_ptr = menu_from_interactive(interactive_ptr);
-
-    menu_ptr = menu_ptr;
-    x = x; y = y;
 
     if (wlr_pointer_button_event_ptr->button != BTN_RIGHT) return;
     switch (wlr_pointer_button_event_ptr->state) {

--- a/src/menu_item.c
+++ b/src/menu_item.c
@@ -89,10 +89,9 @@ void wlmaker_menu_item_destroy(wlmaker_menu_item_t *menu_item_ptr)
 
 /* ------------------------------------------------------------------------- */
 void wlmaker_menu_item_get_desired_size(
-    const wlmaker_menu_item_t *menu_item_ptr,
+    __UNUSED__ const wlmaker_menu_item_t *menu_item_ptr,
     uint32_t *width_ptr, uint32_t *height_ptr)
 {
-    menu_item_ptr = menu_item_ptr;  // currently unused.
     if (NULL != width_ptr) *width_ptr = 256;
     if (NULL != height_ptr) *height_ptr = 22;
 }

